### ISSUE TITLE
W0631 undefined-loop-variable false positive

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#202]
Originally reported by: **Chris Rebert (BitBucket: [cvrebert](http://bitbucket.org/cvrebert), GitHub: @cvrebert?)**

---

Environment:

```
pylint 1.2.0, 
astroid 1.1.0, common 0.61.0
Python 2.7.6 (default, Apr 15 2014, 13:08:46) 
[GCC 4.2.1 Compatible Apple LLVM 5.1 (clang-503.0.40)]
```

Reduced testcase:

```
#!python

#!/usr/bin/env python2.7

def example(args):
    for i in args:
        def nested():
            print i
        nested()
    for i in args:  # this loop is necessary to trigger the bug
        print i
```

pylint output:

```
 $ pylint --disable=C0111 -r n -f parseable example.py 
No config file found, using default configuration
/envs/new-pylint/lib/python2.7/site-packages/pylint/reporters/text.py:79: UserWarning: parseable output format is deprecated. This is equivalent to --msg-template={path}:{line}: [{msg_id}({symbol}), {obj}] {msg}
  'to --msg-template=%s' % (self.name, self.line_format))
************* Module example
example.py:6: [W0631(undefined-loop-variable), example.nested] Using possibly undefined loop variable 'i'
```

The code seems safe AFAICT. No warning is reported if the second loop is removed.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/202
